CMake/soh-cvars.cmake

```
@@ -0,0 +1,26 @@
+set(CVAR_PREFIX_RANDOMIZER_ENHANCEMENT "gRandoEnhancements")
+set(CVAR_PREFIX_RANDOMIZER_SETTING "gRandoSettings")
+set(CVAR_PREFIX_COSMETIC "gCosmetics")
+set(CVAR_PREFIX_AUDIO "gAudioEditor")
+set(CVAR_PREFIX_CHEAT "gCheats")
+set(CVAR_PREFIX_ENHANCEMENT "gEnhancements")
+set(CVAR_PREFIX_SETTING "gSettings")
+set(CVAR_PREFIX_WINDOW "gOpenWindows")
+set(CVAR_PREFIX_TRACKER "gTrackers")
+set(CVAR_PREFIX_DEVELOPER_TOOLS "gDeveloperTools")
+set(CVAR_PREFIX_GENERAL "gGeneral")
+set(CVAR_PREFIX_REMOTE "gRemote")
+add_compile_definitions(
+	CVAR_PREFIX_RANDOMIZER_ENHANCEMENT="${CVAR_PREFIX_RANDOMIZER_ENHANCEMENT}"
+	CVAR_PREFIX_RANDOMIZER_SETTING="${CVAR_PREFIX_RANDOMIZER_SETTING}"
+	CVAR_PREFIX_COSMETIC="${CVAR_PREFIX_COSMETIC}"
+	CVAR_PREFIX_AUDIO="${CVAR_PREFIX_AUDIO}"
+	CVAR_PREFIX_CHEAT="${CVAR_PREFIX_CHEAT}"
+	CVAR_PREFIX_ENHANCEMENT="${CVAR_PREFIX_ENHANCEMENT}"
+	CVAR_PREFIX_SETTING="${CVAR_PREFIX_SETTING}"
+	CVAR_PREFIX_WINDOW="${CVAR_PREFIX_WINDOW}"
+	CVAR_PREFIX_TRACKER="${CVAR_PREFIX_TRACKER}"
+	CVAR_PREFIX_DEVELOPER_TOOLS="${CVAR_PREFIX_DEVELOPER_TOOLS}"
+	CVAR_PREFIX_GENERAL="${CVAR_PREFIX_GENERAL}"
+	CVAR_PREFIX_REMOTE="${CVAR_PREFIX_REMOTE}"
+)
```

Sorry for repeatedly returning to this with new ideas, but since `set()` for normal variables is scope specific, and we're only using these in this file, would you be opposed to changing this to require less boilerplate so we could just have

```suggestion
add_compile_definitions(
	CVAR_PREFIX_RANDOMIZER_ENHANCEMENT="gRandoEnhancements"
	CVAR_PREFIX_RANDOMIZER_SETTING="gRandoSettings"
	CVAR_PREFIX_COSMETIC="gCosmetics"
	CVAR_PREFIX_AUDIO="gAudioEditor"
	CVAR_PREFIX_CHEAT="gCheats"
	CVAR_PREFIX_ENHANCEMENT="gEnhancements"
	CVAR_PREFIX_SETTING="gSettings"
	CVAR_PREFIX_WINDOW="gOpenWindows"
	CVAR_PREFIX_TRACKER="gTrackers"
	CVAR_PREFIX_DEVELOPER_TOOLS="gDeveloperTools"
	CVAR_PREFIX_GENERAL="gGeneral"
	CVAR_PREFIX_REMOTE="gRemote"
)
```

---

They aren't, though. At least, not all of them. I set them up that way because settings, general and window are all used in the LUS CVar customization. That was the main reason they were in `lus-cvars.cmake` to begin with, I just didn't know it would still work if they were included separately.

---

ah, gotcha, nevermind then